core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -786,7 +817,67 @@ private CacheLoader undelegate(CacheLoader l) {
 
    private CacheWriter undelegate(CacheWriter w) {
       return (w instanceof DelegatingCacheWriter) ? ((DelegatingCacheWriter)w).undelegate() : w;
+   }
 
+   private void waitForConnectionInterval() {
+      int connectionInterval = configuration.persistence().connectionInterval();
+      if (connectionInterval > 0) {
+         try {
+            Thread.sleep(connectionInterval);
+         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Same thing here, maybe we should return if it waited full time or was interrupted. I would think we would want to end early if we were interrupted and not do all the connection attempts.